src/Storages/HDFS/HDFSCommon.cpp

```diff
@@ -123,6 +123,12 @@ HDFSBuilderWrapper createHDFSBuilder(const String & uri_str, const Poco::Util::A
     if (host.empty())
         throw Exception("Illegal HDFS URI: " + uri.toString(), ErrorCodes::BAD_ARGUMENTS);
 
+    // Shall set env LIBHDFS3_CONF *before* HDFSBuilderWrapper construction.
+    const String & libhdfs3_conf = config.getString(HDFSBuilderWrapper::CONFIG_PREFIX + ".libhdfs3_conf", "");
```

Comment:
@kssenii Is it necessary to trim `libhdfs3_conf`?

Reply:
Not necessary.